pkg/client/client.go

```diff
@@ -286,21 +309,77 @@ func (f *imageFactory) NewImage(repoName string, daemon bool, imageOS string) (i
 	return remote.NewImage(repoName, f.keychain, remote.WithDefaultPlatform(platform))
 }
 
+func (f *indexFactory) LoadIndex(repoName string, opts ...index.Option) (img imgutil.ImageIndex, err error) {
+	opts, err = withXDGPath(opts, f.keychain)
+	if err != nil {
+		return nil, err
+	}
+
+	img, err = local.NewIndex(repoName, opts...)
+	if err == nil {
+		return
+	}
+
+	img, err = layout.NewIndex(repoName, opts...)
+	if err == nil {
+		return
+	}
+	return nil, errors.Wrap(err, errors.Errorf("Image: '%s' not found", repoName).Error())
+}
+
 type indexFactory struct {
-	dockerClient local.DockerClient
-	keychain     authn.Keychain
+	keychain       authn.Keychain
+	xdgRuntimePath string
 }
 
-func (f *indexFactory) NewIndex(opts CreateManifestOptions) (imgutil.ImageIndex, error) {
-	if opts.Publish {
-		return remote.NewIndex(
-			opts.ManifestName,
-			f.keychain,
-			remote.WithIndexMediaTypes(opts.MediaType))
+func (f *indexFactory) FetchIndex(name string, opts ...index.Option) (idx imgutil.ImageIndex, err error) {
+	opts, err = withXDGPath(opts, f.keychain)
+	if err != nil {
+		return nil, err
+	}
+
+	idx, err = remote.NewIndex(name, opts...)
+	if err != nil {
+		return idx, fmt.Errorf("ImageIndex in not available at registry")
+	}
+
+	return idx, err
+}
+
+func (f *indexFactory) FindIndex(repoName string, opts ...index.Option) (idx imgutil.ImageIndex, err error) {
+	opts, err = withXDGPath(opts, f.keychain)
+	if err != nil {
+		return nil, err
+	}
+
+	idx, err = f.LoadIndex(repoName, opts...)
+	if err == nil {
+		return idx, err
+	}
+
+	return f.FetchIndex(repoName, opts...)
+}
+
+func (f *indexFactory) CreateIndex(repoName string, opts ...index.Option) (imgutil.ImageIndex, error) {
+	opts, err := withXDGPath(opts, f.keychain)
+	if err != nil {
+		return nil, err
+	}
+
+	return index.NewIndex(repoName, opts...)
+}
+
+func withXDGPath(ops []index.Option, keychain authn.Keychain) ([]index.Option, error) {
+	xdgPath, ok := os.LookupEnv(xdgRuntimePath)
+	if ok {
+		ops = append(ops, index.WithKeychain(keychain), index.WithXDGRuntimePath(xdgPath))
+		return ops, nil
+	}
+	home, err := iconfig.PackHome()
+	if err != nil {
+		return ops, err
 	}
 
-	return local.NewIndex(
-		opts.ManifestName,
-		opts.ManifestDir,
-		local.WithIndexMediaTypes(opts.MediaType))
+	ops = append(ops, index.WithKeychain(keychain), index.WithXDGRuntimePath(home))
```

Comment:
I will suggest to keep @husni-faiz  approach here. he was doing this

```go
manifestDir := filepath.Join(packHome, "manifests")
```
basically add a root folder to organize the manifests, we also safe more information in `PackHome` and throwing the manifests there make hard for end users to understand what it is about. 

For example my local .pack home looks like

```bash
> tree ~/.pack -L 1
/Users/jbustamante/.pack
├── cnbs                    <-- this one was created after manifest create
├── completion.zsh
├── config.toml
├── download-cache
├── layout-repo
├── registry-a932275bd19c2d9e1b88fa06698fd2f5427a363d25bf87fa500691c373089381
└── registry2836348280

6 directories, 2 files
```
In pack we used to create a top folder, like `donwload-cache` or `layout-repo`